pkg/core/basic_chain_test.go

```diff
@@ -0,0 +1,595 @@
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/nspcc-dev/neo-go/pkg/config"
+	"github.com/nspcc-dev/neo-go/pkg/core/block"
+	"github.com/nspcc-dev/neo-go/pkg/core/chaindump"
+	"github.com/nspcc-dev/neo-go/pkg/core/dao"
+	"github.com/nspcc-dev/neo-go/pkg/core/native"
+	"github.com/nspcc-dev/neo-go/pkg/core/native/nativenames"
+	"github.com/nspcc-dev/neo-go/pkg/core/storage"
+	"github.com/nspcc-dev/neo-go/pkg/encoding/fixedn"
+	"github.com/nspcc-dev/neo-go/pkg/interop/native/roles"
+	"github.com/nspcc-dev/neo-go/pkg/io"
+	"github.com/nspcc-dev/neo-go/pkg/neotest"
+	"github.com/nspcc-dev/neo-go/pkg/neotest/chain"
+	"github.com/nspcc-dev/neo-go/pkg/rpc/client/nns"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+	"github.com/nspcc-dev/neo-go/pkg/wallet"
+	"github.com/stretchr/testify/require"
+)
+
+var notaryModulePath = filepath.Join("..", "services", "notary")
+
+// examplesPrefix is a prefix of the example smart-contracts.
+const examplesPrefix = "../../examples/"
+
+// This function generates "../rpc/testdata/testblocks.acc" file which contains data
+// for RPC unit tests. It also is a nice integration test.
+// To generate new "../rpc/testdata/testblocks.acc", follow the steps:
+// 		1. Set saveChain down below to true
+// 		2. Run tests with `$ make test`
+func TestCreateBasicChain(t *testing.T) {
+	const saveChain = false
+	const prefix = "../rpc/server/testdata/"
+
+	bc, validators, committee := chain.NewMultiWithCustomConfig(t, func(cfg *config.ProtocolConfiguration) {
+		cfg.P2PSigExtensions = true
+	})
+	e := neotest.NewExecutor(t, bc, validators, committee)
+
+	initBasicChain(t, e)
+
+	if saveChain {
+		outStream, err := os.Create(prefix + "testblocks.acc")
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			outStream.Close()
+		})
+
+		writer := io.NewBinWriterFromIO(outStream)
+		writer.WriteU32LE(bc.BlockHeight())
+		err = chaindump.Dump(bc, writer, 1, bc.BlockHeight())
+		require.NoError(t, err)
+	}
+
+	require.False(t, saveChain)
+}
+
+func initBasicChain(t *testing.T, e *neotest.Executor) {
+	if !e.Chain.GetConfig().P2PSigExtensions {
+		t.Fatal("P2PSitExtensions should be enabled to init basic chain")
+	}
+	const prefix = "../rpc/server/testdata/"
+
+	const neoAmount = 99999000
+
+	gasHash := e.NativeHash(t, nativenames.Gas)
+	neoHash := e.NativeHash(t, nativenames.Neo)
+	policyHash := e.NativeHash(t, nativenames.Policy)
+	notaryHash := e.NativeHash(t, nativenames.Notary)
+	designationHash := e.NativeHash(t, nativenames.Designation)
+	t.Logf("native GAS hash: %v", gasHash)
+	t.Logf("native NEO hash: %v", neoHash)
+	t.Logf("native Policy hash: %v", policyHash)
+	t.Logf("native Notary hash: %v", notaryHash)
+	t.Logf("Block0 hash: %s", e.Chain.GetHeaderHash(0).StringLE())
+
+	acc0 := e.Validator.(neotest.MultiSigner).Single(2) // TODO: port index->order and order->index conversion to neotest?
```

Comment:
OK, TODO is removed. But I still prefer no to create a separate account for this particular test because ew need to retrieve it in our RPC tests (and with priv0 we know it exatly and don't need additional steps to retrieve its address).